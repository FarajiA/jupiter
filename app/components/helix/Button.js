import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  render() {
    const { label, disabled, classNames, type, onClick, children } = this.props;
    return (
      <button className={`hxBtn ${classNames}`} type={type} disabled={disabled} onClick={onClick}>
        {label}
        {children}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNames: PropTypes.string,
  type: PropTypes.string
};

Button.defaultProps = {
  disabled: false,
  classNames: '',
  type: 'button'
};


export default Button;
