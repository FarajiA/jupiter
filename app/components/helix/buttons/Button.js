import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  render() {
    const { label, disabled, classNames, onClick, processing, submit, children } = this.props;
    return (
      <button // eslint-disable-line react/button-has-type
        className={`hxBtn ${classNames}`}
        type={submit ? 'submit' : 'button'}
        disabled={disabled}
        onClick={onClick}
      >
        {!processing && (label || children)}
        {processing && <span>{label || children}</span>}
        {processing && <hx-busy />}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNames: PropTypes.string,
  processing: PropTypes.bool,
  submit: PropTypes.bool
};

Button.defaultProps = {
  disabled: false,
  classNames: '',
  submit: false
};


export default Button;
