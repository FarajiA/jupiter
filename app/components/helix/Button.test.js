import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  const props = {
    label: 'Click Me!'
  };
  test('it renders correct props', () => {
    const component = renderer.create(<Button {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Label prop is rendered', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper.text()).toEqual('Click Me!');
  });
});
