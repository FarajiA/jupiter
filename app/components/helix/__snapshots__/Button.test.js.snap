// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it renders correct props 1`] = `
<button
  className="hxBtn "
  disabled={false}
  type="button"
>
  Click Me!
</button>
`;
